tools/run_tests/artifacts/artifact_targets.py

```diff
@@ -403,6 +419,16 @@ def targets():
         PythonArtifact('linux_extra', 'armv7', 'cp38-cp38'),
         PythonArtifact('linux_extra', 'armv7', 'cp39-cp39'),
         PythonArtifact('linux_extra', 'armv7', 'cp310-cp310', presubmit=True),
+        PythonArtifact('musllinux_1_1', 'x64', 'cp310-cp310', presubmit=True),
```

Comment:
given all the recent changes/speedup in the build_artifacts related jobs, I'd like to see a summary on how much extra time these artifacts add  on presubmit and on master.
I really don't want to annulate all the speedup we just did by adding 10 new artifacts.

If the numbers don't look good, we'll need to come up with more ideas how to speed things up on master and presubmit.

Reply:
In the description of https://github.com/grpc/grpc/pull/28228, I updated the latest time spend on each job. This PR should only affect the Linux artifact build job and the Python distribtest job. The time spans in parenthesis order in: before optimization, after optimization, this PR.

- grpc/core/pull_request/linux/grpc_build_artifacts (170 mins -> 68 mins -> 79 mins)
- grpc/core/pull_request/linux/grpc_distribtests_python (120 mins -> 29mins -> 54 mins)

They are not the slowest presubmit checks as of today.

Reply:
I'd like us to be targeting <20 minutes across the board for pull requests.

Reply:
Well, we need significant improvement in the infrastructure to achieve this. In past 2 weeks, Jan and I have reached for low hanging fruits we can find, and the result is reducing the build time to 1/3 or 1/2.

Feel free to share your thought on go/grpc-ci-options. We are discussing the next step of our infrastructure.

Reply:
I agree with Craig that we should be more aggressive in terms of reducing the duration of PR jobs. It would be nice if we could come up with more speedups and only then added more stuff that makes the builds slower.

Reply:
FTR, I think the grpc_build_artifacts slowdown potentially added by this PR is partially masked by the fact that the all the ruby artifacts are built serially (in a single 'task') and the long tail of the ruby artifact build makes to job finish much later than it otherwise could. I'm planning to remove that limitation in https://github.com/grpc/grpc/pull/28243, and once that PR is in, the negative effect of adding the musllinux artifacts will become more significant. I don't think we want that.

Reply:
@jtattermusch What can I do to push this PR forward?